tests/functional/Tribe/Events/Navigation_Test.php

```diff
@@ -0,0 +1,59 @@
+<?php
+namespace Tribe\Events;
+
+class Navigation_Test extends \Tribe__Events__WP_UnitTestCase {
+	public function setUp() {
+		// before
+		parent::setUp();
+		// your set up methods here
+	}
+
+	public function tearDown() {
+		// your tear down methods here
+		// then
+		parent::tearDown();
+	}
+
+	public function test_closest_event() {
```

Comment:
I've added a `test_closest_event_non_linear()` method that sets up the data in the same way you suggested and it passes!